test/test_compile_test_and_run.cpp

```diff
@@ -0,0 +1,39 @@
+//  Copyright John Maddock 2014.
+//  Copyright Christopher Kormanyos 2024.
+//  Use, modification and distribution are subject to the
+//  Boost Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <boost/core/lightweight_test.hpp>
+#include <boost/cstdfloat.hpp>
+
+#include "compile_test/instantiate.hpp"
+
+namespace local
+{
+  auto instantiate_runner() -> void
+  {
+    #if defined(BOOST_FLOAT64_C)
+
+    instantiate(static_cast<boost::float64_t>(BOOST_FLOAT64_C(1.23)));
+
+    const bool result_instantiate_and_run_is_ok = instantiate_runner_result<boost::float64_t>::value;
+
+    BOOST_TEST(result_instantiate_and_run_is_ok);
+
+    #else
+
+    BOOST_TEST(true);
+
+    #endif
+  }
+}
+
+auto main() -> int
+{
+  local::instantiate_runner();
+
+  const auto result_is_ok = (boost::report_errors() == 0);
+
+  return (result_is_ok ? 0 : -1);
```

Comment:
Silly question: why not just return `boost::report_errors()` like everyone else? ;)

Reply:
> Silly question: why not just return `boost::report_errors()`

We can work out these syntactic details. Thanks.

> Does this increase code coverage?

Yes it increases code coverage.

> If so, that's arguably a bad thing.

To me this doesn't make sense. I get that there can be compile checks. But when the arguments are fixed, the code should run. Maybe I missed some syntactic details when I got the code running. Then I made a few mistakes (maybe) and the code does not compile. But i think it should do both.

We can hash out the syntactic details later, but I think the code in instantiate.hpp should both compile and run. Sure this might be a bit of a shocker in legacy code, but I would have a difficult time understanding why compileing exlcudes running (as long as the arguments are in range).